core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts

```diff
@@ -0,0 +1,146 @@
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+
+import { MenuItem } from 'primeng/api';
+import { ButtonModule } from 'primeng/button';
+import { MenuModule } from 'primeng/menu';
+
+import { ActionPayload } from '../../../shared/models';
+import { ContentletArea } from '../ema-page-dropzone/ema-page-dropzone.component';
+
+@Component({
+    selector: 'dot-ema-contentlet-tools',
+    standalone: true,
+    imports: [CommonModule, ButtonModule, MenuModule],
+    templateUrl: './ema-contentlet-tools.component.html',
+    styleUrls: ['./ema-contentlet-tools.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class EmaContentletToolsComponent {
+    private buttonPosition: 'after' | 'before' = 'after';
+
+    @Input() contentlet: ContentletArea;
+    @Output() addContent = new EventEmitter<ActionPayload>();
+    @Output() addForm = new EventEmitter<ActionPayload>();
+    @Output() addWidget = new EventEmitter<ActionPayload>();
+    @Output() edit = new EventEmitter<ActionPayload>();
+    @Output() delete = new EventEmitter<ActionPayload>();
```

Comment:
This is all emitting the same. Should we have one event and send a type to identify the action?

Reply:
I tried that first but it sucked, harder to test, bigger handlers etc...